### PR TITLE
[FIX] fieldservice: Share stages/teams between companies

### PR DESCRIPTION
**Pull request by osi-scampbell**
This PR allows companies to share Stages and Teams so the base workflow can transition between companies as well

### CODE REVIEW

**Review by max3903 — APPROVED** [review]

File: fieldservice/models/fsm_order.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ class FSMOrder(models.Model):
     def _default_stage_id(self):
         stage_ids = self.env['fsm.stage'].\
             search([('stage_type', '=', 'order'),
-                    ('company_id', '=', self.env.user.company_id.id)],
+                    ('company_id', '=', self.env.user.company_id.id or False)],
```
Comment (max3903) [inline fieldservice/models/fsm_order.py]:
```suggestion
                    ('company_id', '=', self.env.user.company_id.id or False),
                    ('is_default', '=', True)],
```

File: fieldservice/models/fsm_order.py [hunk 2]
```diff
@@ -25,8 +26,8 @@ def _default_stage_id(self):
 
     def _default_team_id(self):
         team_ids = self.env['fsm.team'].\
-            search([('company_id', '=', self.env.user.company_id.id)],
-                   order='sequence asc')
+            search([('company_id', '=', self.env.user.company_id.id or False)],
+                    order='sequence asc')
```
Comment (max3903) [inline fieldservice/models/fsm_order.py]:
```suggestion
                   order='sequence asc')
```

**Review by patrickrwilson — APPROVED** [review]

File: fieldservice/models/fsm_order.py [hunk 3]
```diff
@@ -15,23 +15,28 @@ class FSMOrder(models.Model):
...
-                                    FSM Order Stage first."))
+            raise ValidationError(_(
+                "You must create an FSM order stage first."))
 
     def _default_team_id(self):
         team_ids = self.env['fsm.team'].\
-            search([('company_id', '=', self.env.user.company_id.id)],
+            search(['|',
+                    ('company_id', '=', self.env.user.company_id.id),
+                    ('company_id', '=', False)],
```
Comment (hparfr) [inline fieldservice/models/fsm_order.py]: you can use `["company_id", "in" , (self.env.user.company_id.id, False)]` to be more readable

File: fieldservice/models/fsm_order.py [hunk 4]
```diff
@@ -15,23 +15,28 @@ class FSMOrder(models.Model):
     def _default_stage_id(self):
         stage_ids = self.env['fsm.stage'].\
             search([('stage_type', '=', 'order'),
-                    ('company_id', '=', self.env.user.company_id.id)],
+                    ('is_default', '=', True),
+                    '|',
+                    ('company_id', '=', self.env.user.company_id.id),
+                    ('company_id', '=', False)],
                    order='sequence asc')
```
Comment (hparfr) [inline fieldservice/models/fsm_order.py]: add a limit=1

**Review by hparfr — APPROVED** [review]
Minor non-blocking remarks.

**Pull request merged**